task/mgmt-gateway/src/update/mod.rs

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::mgs_handler::BufferMutex;
+use gateway_messages::{
+    ComponentUpdatePrepare, ResponseError, UpdateId, UpdateStatus,
+};
+
+#[cfg(feature = "gimlet")]
+pub(crate) mod host_flash;
+
+mod common;
+pub(crate) mod sp;
+
+// TODO Currently we only have one implementor of this trait
+// (`HostFlashUpdate`), and we don't actually use the trait directly anywhere.
+// This is currently just a guide for how we expect component update
+// implementations to be structured; once we have more than one, we can adjust
+// this trait definition and/or see if it makes sense to have some generic
+// support code.
+//
+// `SpUpdate` has methods with similar names/signatures, but aren't quite the
+// same due to having to deal with both auxflash and SP images. It therefore
+// does not implement this trait. That's another wart that we might be able to
+// address once we start adding more implementors and tweak this trait.
+pub(crate) trait ComponentUpdater {
+    /// Size of one block / sector / page; whatever unit the underlying update
+    /// mechanism wants as a single chunk.
+    fn block_size(&self) -> usize;
+
+    /// Attempt to start preparing for an update, using `buffer` as the backing
+    /// store for incoming data.
```

Comment:
Add a comment here that this records the `UpdateId` from `ComponentUpdatePrepare`, since that wasn't obvious to me